### ISSUE TITLE
Look into possibility of buffer time battery for Pi

### ISSUE THREAD

**Issue by PaoloGB** [Rymapt/ethoscope#20]

**Comment by PaoloGB** [comment]
Closed because already discussed in #18.

**Comment by mark-grimes** [comment]
I'd say they're different issues, although all the discussion on #18 so far should have been here.  #18 is for software to set the time on the ethoscope from the computer with the browser.  This is for hardware.  We want both since even with a hardware clock we need to be able to set a reference point. #18 is also more urgent.

**Comment by PaoloGB** [comment]
I agree. I messed up my first line of posts but we can continue the hardware discussion here.

I have ordered teo CRTs to make some tests. They should arrive in the next couple of days (at my place, since they should fit through the post).

**Comment by PaoloGB** [comment]
Tested RTC chip today.
I managed to talk to it and write/read the important registers.
For some reason it complains about the oscillator not being ok, so at the moment it does not see time passing: once the date is set, it remains at that value.
I need to tidy up my code, then I can merge it. Meanwhile I will try to figure out the hardware issue.

**Comment by PaoloGB** [comment]
Actions:

- add 47 kohm to IRQ/FOUT pin to Vcc. When enabled, this should show the oscillator.
- add decoupling capacitor between Vcc and Gnd

If still not working, buy an oscillator from list on page 20 of [RTC datasheet](http://docs-europe.electrocomponents.com/webdocs/151e/0900766b8151e5f9.pdf).

**Comment by PaoloGB** [comment]
I tried the modifications suggested in my previous post but still no luck.
Checking the datasheet, it looks like I bought an oscillator that is not compatible with the requirement.
I have ordered a [replacement quartz](http://uk.rs-online.com/web/p/crystal-units/9047409/?searchTerm=9047409&relevancy-data=636F3D3126696E3D4931384E525353746F636B4E756D626572266C753D656E266D6D3D6D61746368616C6C26706D3D5E5C647B362C377D7C5C647B31307D7C28283F69292852537C5253207C52532D293F5C647B337D285C73293F5B5C732D2F255C2E2C5D285C73293F5C647B332C347D292426706F3D3126736E3D592673743D52535F53544F434B5F4E554D4245522677633D4E4F4E45267573743D39303437343039267374613D3930343734303926) that should be here tomorrow.

**Comment by PaoloGB** [comment]
I got the RTC working.
Tested this morning with a 3 V battery:
- set date/time
- check the seconds and minutes are increasing
- power off the Pi, disconnect the cable
- booted the Pi again, the time was still ok

I have a Pythin script to perform basic operations on the chip (set time, read time, read status).
Need to look how to make the Pi pick up the time automatically at boot.
[This page](https://thepihut.com/blogs/raspberry-pi-tutorials/17209332-adding-a-real-time-clock-to-your-raspberry-pi) seems a good start.
Also [this](http://spellfoundry.com/sleepy-pi/accessing-real-time-clock-raspberry-pi/).

**Comment by PaoloGB** [comment]
Test:
set date time correct at 11:22:45 on 4/7/2017.
Powered off the Pi at 11:24:30.

Powered on at 15:00, the time is still correct to the second.
Powered off again at 15:02.
Powered at 09:48 on 5/7/2017, the clock is still fine. About 3 seconds off between my computer and the RTC.
